firedrake/mg/embedded.py

```diff
@@ -87,14 +87,17 @@ def _native_transfer(self, element, op):
         return None
 
     def cache(self, V):
+        mh, _ = get_level(V.mesh())
+        caches = mh._shared_data_cache["transfer_manager_cache"]
         key = (V.ufl_element(), V.value_shape, V.boundary_set)
         try:
-            return self.caches[key]
+            return caches[key]
         except KeyError:
-            return self.caches.setdefault(key, TransferManager.Cache(*key[:2]))
+            return caches.setdefault(key, TransferManager.Cache(*key[:2]))
```

Comment:
```suggestion
            return caches.setdefault(key, TransferManager.Cache(*key[:3]))
```
? If so, `TransferManager.Cache.__init__()` must be updated. I should have asked this in the previous PR.